### PR TITLE
[BEAM-4744] Enable parallel publishing/release

### PR DESCRIPTION
**Pull request by lukecwik**
This solves the problem where jars would overwrite each other when specifying -Ppublishing by fixing classifiers for shadowJar and jar to not conflict when publishing since jar by default uses null.

Note that I also fixed:
* some permissions issues of files
* javadoc issues that came up when I was attempting to publish
* the usage of :runners:google-cloud-dataflow-java:examples and examples-streaming. This was creating an empty project that was being published which was causing issues when attempting to perform a release.

I was able to run a publishToMavenJavaPublicationToMavenLocal in parallel but didn't explicitly enable all the publishing plugins and remove the need for -Ppublishing because I didn't get enough time to test it. This cut publishing time down to about 8 mins from over 20 mins in a clean workspace on my machine.

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_GradleBuild/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_GradleBuild/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark_Gradle/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | --- | --- | --- | ---

### CODE REVIEW

**Comment by lukecwik** [comment]
R: @apilloud @pabloem 
CC: @boyuanzz @alanmyrvold

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy [hunk 1]
```diff
@@ -663,9 +662,14 @@ class BeamModulePlugin implements Plugin<Project> {
...
 
+      project.jar {
+        classifier = "unshaded"
+        zip64 true
+      }
+
       // Always configure the shadowJar classifier and merge service files.
       project.shadowJar({
-        classifier = "shaded"
+        classifier = null
```
Comment (apilloud) [inline buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy]: This will make @kennknowles happy I think.

**Comment by lukecwik** [comment]
Run Java Precommit

**Comment by lukecwik** [comment]
Run Java Precommit

**Comment by udim** [comment]
I added a special `publish` property to JavaNatureConfiguration so that those examples project would not be published.
https://github.com/apache/beam/blob/f0fb7f67a54f815c9342adc87749ed0b7d962187/buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy#L737

What empty project was being published? I'm curious.

**Comment by lukecwik** [comment]
@udim Did you ask this on the right PR?

**Comment by udim** [comment]
Yeah, this is regarding your changes in settings.gradle and the comment:

> the usage of :runners:google-cloud-dataflow-java:examples and examples-streaming. This was creating an empty project that was being published which was causing issues when attempting to perform a release.

**Pull request merged**